src/main/java/org/cloudfoundry/promregator/auth/TokenFetcher.java

```diff
@@ -0,0 +1,170 @@
+package org.cloudfoundry.promregator.auth;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.http.Consts;
+import org.apache.http.HttpEntity;
+import org.apache.http.NameValuePair;
+import org.apache.http.ParseException;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.http.util.EntityUtils;
+import org.cloudfoundry.promregator.auth.OAuth2XSUAAEnricher.TokenResponse;
+import org.cloudfoundry.promregator.config.OAuth2XSUAAAuthenticationConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.gson.Gson;
+
+public abstract class TokenFetcher {
```

Comment:
Also here: Name of class sounds a little too generic to me; perhaps in its own package, this would be better as well.

Reply:
Will not be present any more. Still exists at the moment, but contains only a main method for real-world-testing. The class will be dropped before merging.

Reply:
This class will be dropped (there is at the moment only a main method for real-world-testing.